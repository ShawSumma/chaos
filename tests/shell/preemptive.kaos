void def f4()
    print b
end

function_table

num def f1()
    num a = 101
    return a
end

num def f2()
    num b = 102
    return b
end

num def f3()
    num c = 103
    return c
end

num def f4(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 8  : f1(),
    n1 > 10 : f2(),
    default : f3()
}

function_table


num def f4(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 8  : f1(),
    z > 10  : f2(n2),
    default : f3()
}

function_table


void def a()
    num b = f5()
end

function_table


num def f4(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 8  : f1(),
    z > 10  : f6(),
    default : f3()
}

function_table


num def f4(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 8  : f1(z),
    z > 10  : f2(),
    default : f3()
}

function_table


num def f4()
    num c = 103
    num x = f1(c)
    return c
end

function_table


num def f8(num a, str b)
    print a
    print b
    num c = a + 5
    return c
end

function_table

void def f4()
    num a = 3
    num b = 5
    f8(a, b)
end

function_table

void def f4()
    f8(3, 5)
end

function_table

void def f4()
    list a = [1, 2, 3]
    str b = 'hello'
    f8(a, b)
end

function_table

void def f4()
    f8([1, 2, 3], 'hello')
end

function_table

void def f4()
    num a = 3
    str b = 'hello'
    bool c = true
    f8(a, b, c)
end

function_table

void def f4()
    f8(3, 'hello', true)
end

function_table


list def f4()
    dict a = {'a': 1, 'b': 2, 'c': 3}
    return a
end

function_table

dict def f4()
    dict a = {'a': 1, 'b': 2, 'c': 3}
    return b
end

function_table


list def f4()
    num x = 5
    dict a = {'a': 1, 'b': 2, 'c': 3}
end {
    x == 5  : return a,
    default : return a
}

function_table

dict def f4()
    num x = 5
    dict a = {'a': 1, 'b': 2, 'c': 3}
end {
    x == 5  : return b,
    default : return b
}

function_table

dict def f4()
    num x = 5
end

function_table


void def f4()
    3 times do
        print b
        2 times do
            print c
        end
    end
end

function_table

void def f4()
    n times do
        print "hello world"
    end
end

function_table

void def f4()
    foreach a as el
        print el
    end
end

function_table

void def f7()
    list a = [1, 2, 3]
    foreach a as el
        print el
    end
end

function_table

void def f4()
    list a = [1, 2, 3]
    foreach a as el
        print el
        print b
    end
end

function_table

void def f8()
    dict a = {'a': 1, 'b': 2, 'c': 3}
    foreach a as key : val
        print key
        print val
    end
end

function_table

void def f4()
    dict a = {'a': 1, 'b': 2, 'c': 3}
    foreach a as key : val
        print key
        print val
        print b
    end
end

function_table


num def f9(num x, num y)
    num z = x + y
    bool a = true
end {
    z > 5   : break,
    z == 8  : f1(),
    z > 10  : f2(),
    default : f3()
}

function_table

void def f4(num list a)
    foreach a as el
        print el
        print f11(el, 1)
    end
end

function_table

void def f4(num a)
    foreach a as el
        print el
        print f9(el, 1)
    end
end

function_table

void def f4(num a)
    foreach a as key : val
        print key
        print f9(val, 1)
    end
end

function_table
