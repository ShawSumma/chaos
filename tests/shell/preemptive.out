[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 2, in f4
[0;43m      print b
[1;43m  Undefined variable: b
Absorbed by Interactive Shell
[start] =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 27, in f4
[0;43m      n1 > 10 : f2(),
[1;43m  Undefined variable: n1
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 39, in f4
[0;43m      z > 10  : f2(n2),
[1;43m  Undefined variable: n2
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 47, in a
[0;43m      num b = f5()
[1;43m  Undefined function: f5 in <module>
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 58, in f4
[0;43m      z > 10  : f6(),
[1;43m  Undefined function: f6 in <module>
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 69, in f1
[0;43m      z == 8  : f1(z),
[1;43m  Incorrect argument count for function: f1
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 79, in f1
[0;43m      num x = f1(c)
[1;43m  Incorrect argument count for function: f1
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 98, in f8
[0;43m      f8(a, b)
[1;43m  Illegal variable type for function parameter: b of function: f8
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 104, in f8
[0;43m      f8(3, 5)
[1;43m  Illegal variable type for function parameter: b of function: f8
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 112, in f8
[0;43m      f8(a, b)
[1;43m  Illegal variable type for function parameter: a of function: f8
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 118, in f8
[0;43m      f8([1, 2, 3], 'hello')
[1;43m  Illegal variable type for function parameter: a of function: f8
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 127, in f8
[0;43m      f8(a, b, c)
[1;43m  Incorrect argument count for function: f8
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 133, in f8
[0;43m      f8(3, 'hello', true)
[1;43m  Incorrect argument count for function: f8
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 141, in f4
[0;43m      return a
[1;43m  Illegal variable type: Dictionary for function: f4
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 148, in f4
[0;43m      return b
[1;43m  Undefined variable: b
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 158, in f4
[0;43m      x == 5  : return a,
[1;43m  Illegal variable type: Dictionary for function: f4
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 168, in f4
[0;43m      x == 5  : return b,
[1;43m  Undefined variable: b
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 176, in f4
[0;43m      end
[1;43m  The function 'f4' did not return anything!
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 183, in f4
[0;43m      print b
[1;43m  Undefined variable: b
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 193, in f4
[0;43m      n times do
[1;43m  Undefined variable: n
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 201, in f4
[0;43m      foreach a as el
[1;43m  Undefined variable: a
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 221, in f4
[0;43m      print b
[1;43m  Undefined variable: b
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 242, in f4
[0;43m      print b
[1;43m  Undefined variable: b
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 264, in f4
[0;43m      print f11(el, 1)
[1;43m  Undefined function: f11 in <module>
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 271, in f4
[0;43m      foreach a as el
[1;43m  'a' is not a list!
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 280, in f4
[0;43m      foreach a as key : val
[1;43m  'a' is not a dictionary!
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 290, in f9
[0;43m      num r = f9(x, y)
[1;43m  Call to a function with `break` from outside a loop: f9
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
[1;43m  Preemptive Error:
File: "~/chaos/__interactive__.kaos", line 299, in f9
[0;43m      print f9(el, 1)
[1;43m  Call to a function with `break` from a multiline loop: f9
Absorbed by Interactive Shell
[start] =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
File: "~/chaos/__interactive__.kaos, module: } =>
[end]
103
103
103
103
